src/provider/application.js

```diff
@@ -72,7 +73,32 @@ class Application extends EventEmitter {
       const width = calculateWidth(this.resizeConfig.WidthCalculationMethod);
       this.JSONRPC.notification('resize', [null, `${width}px`]);
     } else {
-      const height = calculateHeight(this.resizeConfig.heightCalculationMethod);
+      let height = calculateHeight(this.resizeConfig.heightCalculationMethod);
+
+      // If targetSelectors is specified from Provider or Consumer or both,
+      // need to calculate the height based on specified target selectors
+      if (this.targetSelectors.length > 0 || this.resizeConfig.targetSelectors) {
+        let targetSelectors = [];
+
+        if (this.targetSelectors.length > 0) {
+          targetSelectors = targetSelectors.concat(this.targetSelectors);
+        }
+        if (this.resizeConfig.targetSelectors) {
+          targetSelectors = targetSelectors.concat(this.resizeConfig.targetSelectors);
+        }
+
+        // Remove duplicates
+        targetSelectors = targetSelectors.filter(
+          (elem, index, arr) => arr.indexOf(elem) === index
+        );
+
+        height = targetSelectors
+          .map(selector => document.querySelectorAll(selector))
+          .reduce((accum, nodes) => [...accum, ...nodes], [])
+          .map(node => (!node ? 0 : getOffsetToBody(node) + node.offsetHeight))
```

Comment:
IMO its cleaner if you handle all of this in the function itself so you just do `.map(getOffsetToBody)`. You can handle the recursion internal in that function and handle the null condition and add offset there.